conanfile.py

```diff
@@ -42,15 +40,9 @@ def export_sources(self):
         shutil.copy("CMakeLists.txt", self.export_sources_folder)
         shutil.copy("LICENSE", self.export_sources_folder)
 
-    def config_options(self):
-        if self.settings.os == "Windows":
-            del self.options.fPIC
-
     def configure(self):
-        if not self.options.static:
```

Comment:
For the recipe at cci, you will have to throw a `ConanInvalidConfiguration` here to disable building on Macos and Windows.
I wouldn't do it here, because it is already disabled in the cmake build script, and I would avoid the double check.

Reply:
Also, maybe add a `package_info` to the class, for completeness.
You will have to add something like this:
```python
    def package_info(self):
        # Set the name of the generated `FindSeasocks.cmake` and `SeasocksConfig.cmake` cmake scripts
        self.cpp_info.names["cmake_find_package"] = "Seasocks"
        self.cpp_info.names["cmake_find_package_multi"] = "Seasocks"
        self.cpp_info.components["libseasocks"].libs = ["seasocks"]
        # Set the name of the generated seasocks target: `Seasocks::seasocks`
        self.cpp_info.components["libseasocks"].names["cmake_find_package"] = "seasocks"
        self.cpp_info.components["libseasocks"].names["cmake_find_package_multi"] = "seasocks"
```
This complexity is needed because conan will by default create a `Findseasocks.cmake` find script, but we want it capitalized. That's the reason of the first set of `self.cpp_info.names`.

The second set is required to change the name of the seasocks target from `Seasocks::libseasocks` back to `Seasocks::seasocks`.
Using the name `seasocks` for a component is disallowed because it is the name of the recipe.

(Components allow to only link to certain libs instead of all libs of a recipe)

Reply:
I literally don't understand any of what you said there :blush: -- it's...all so complex. `-lseasocks` is what I use elsewhere :|

Reply:
The original purpose of conponents is for recipes having multiple libraries, such as boost and openssl.
e.g. openssl: that has a crypto and ssl library (ssl depends on crypto).
So you would create a crypto component and ssl component. That way people only needing crypto can link to that.

In short, all this complexity is to make sure a `Seasocks::seasocks` target is created in a `FindSeasocks.cmake` script by conan.

btw, the snippet above contained a bug. You will have to add:
```python
        if self.options.with_zlib:
            self.cpp_info.components["libseasocks"].requires = ["zlib::zlib"]
```

Reply:
How might I test your suggestion?

Reply:
(and apologies for sounding grumpy about this! :) I so appreciate your patience hand-holding here)

Reply:
> How might I test your suggestion?

You'll have to write a test_package for that, using conan's `cmake_find_package` generator, and link your test application to `Seasocks::seasocks`.

> (and apologies for sounding grumpy about this! :) I so appreciate your patience hand-holding here)

I didn't have that impression. I understand your feeling when seeing this many lines for such a small thing.

Reply:
Thanks! I've added the code as you suggested! I'll add a test_package too (or try to). I have one in the (pending) PR to conan-center

Reply:
Nice, maybe the review process on cci will give some suggestions to simplify this.